src/formatDate.js

```diff
@@ -50,7 +50,39 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const ourDate = date.split(fromFormat[3]);
+  const outputDate = [];
+
+  const ourFormat = {
+    DD: null,
+    MM: null,
+    YY: null,
+    YYYY: null,
+  };
+
+  for (let i = 0; i < 3; i++) {
+    ourFormat[fromFormat[i]] = ourDate[i];
+  }
+
+  // we check if we don't already have the format we need
+  if (toFormat.includes('YY') && ourFormat.YY === null) {
```

Comment:
is there a reason for checking for exactly `null` value? Why not we just checking whether it is not falsy?
```suggestion
  if (toFormat.includes('YY') && !ourFormat.YY) {
```